scrapy/downloadermiddlewares/uriuserinfo.py

```diff
@@ -0,0 +1,49 @@
+from urllib.parse import unquote, urlunparse
+
+from scrapy.utils.httpobj import urlparse_cached
+
+
+class UriUserinfoMiddleware(object):
+    """Downloader middleware that replaces `URI userinfo`_ data (user credentials
+    for HTTP or FTP specified in the request URL) with the corresponding meta
+    keys for later middlewares or download handlers to use them for
```

Comment:
I might be missing something, but AFAICT this won't be enough to authenticate HTTP requests since the current [HTTP/1.1 download handler](https://github.com/scrapy/scrapy/blob/2.0.1/scrapy/core/downloader/handlers/http11.py) does not rely on `meta` keys and the `HttpAuthMiddleware` takes the [credentials from the spider](https://github.com/scrapy/scrapy/blob/2.0.1/scrapy/downloadermiddlewares/httpauth.py#L22-L26), not the request.

Reply:
I updated the documentation but forgot to implement those changes :facepalm: